### DISCUSSION TITLE
Translations field interface improvements

### DISCUSSION THREAD

**Discussion by bposman** [directus/directus#10478] [Feature Requests]
My project requires a few changes in how the translations field interface works, especially in regards to default language selection.

Before I create a custom version of the translations interface as an extension, I wanted to see if it would be possible to get some of the changes into core first. 

I am willing to put some work into this and I'm already part of the way there, but I would also like to hear other's opinions before I think about creating a pull request.

The main changes I would like are:

- Allow a sort field to be specified in the interface settings so I can dictate the order of languages in the translate dropdown.
- Allow a default language to be chosen in a similar way to the translations display.
- Add a checkbox to enable/disable retrieving default language from user (similar to the translations display).
- Add default language fallback to the above, so a user with the language field of `en-GB`, will default to `en` if their own language is not present in the language table.
- A way to chose how the split translation view works with the above, e.g. should default language be on left and user on right, or vice versa (this part I'm least clear about how to implement concisely)

**Comment by rijkvanzanten** [comment]
Good points! 

Here's my quick 2¢

> Allow a sort field to be specified in the interface settings so I can dictate the order of languages in the translate dropdown.

Could we base this on the (manual) sort value of the related languages collection? By adding a sort field on the languages collection, you could drag and drop the manual sort which might give you an even greater level of control 🤔 

> Allow a default language to be chosen in a similar way to the translations display.

+1

> Add a checkbox to enable/disable retrieving default language from user (similar to the translations display).

+1

> Add default language fallback to the above, so a user with the language field of en-GB, will default to en if their own language is not present in the language table.

This is a little tricky. Once you start relying on a "starts with", it becomes increasingly unreliable 🤔 What would we match on? Will we assume the language primary key has a hyphen in it?

> A way to chose how the split translation view works with the above, e.g. should default language be on left and user on right, or vice versa (this part I'm least clear about how to implement concisely)

I'm not sure if that's super useful for most (80/20) users 🤔 Might be a tad overkill.

**Reply by bposman** [reply]
Thanks for the feedback.

>> Allow a sort field to be specified in the interface settings so I can dictate the order of languages in the translate dropdown.
> 
> Could we base this on the (manual) sort value of the related languages collection? By adding a sort field on the languages collection, you could drag and drop the manual sort which might give you an even greater level of control 🤔

Yes, this is what i'm going for in my initial poc, and seems to work pretty well so far.
 
>> Add default language fallback to the above, so a user with the language field of en-GB, will default to en if their own language is not present in the language table.
> 
> This is a little tricky. Once you start relying on a "starts with", it becomes increasingly unreliable 🤔 What would we match on? Will we assume the language primary key has a hyphen in it?

Well, the lookup key for the language is determined by the user's language field (system-language), and at least when using the admin the choices always have hyphens in?  So my poc splits on the hyphen and then just looks up the actual language code ('en', 'es', etc.) in the languages collection.

Just a little note on my use case which is an internal app, we have people internationally in the UK and North America, (similar with Spain and Latin America) but at this point we don't have the resources or need to translate everything into both en-GB and en-US. 

Given that we don't seem to be able to change the user language choices as they are hard coded and dictate the localisation of the admin app, rather than forcing all users to be either en-GB or en-US, I thought the next best thing was to be able to look up the language part only. 

> 
> > A way to chose how the split translation view works with the above, e.g. should default language be on left and user on right, or vice versa (this part I'm least clear about how to implement concisely)
> 
> I'm not sure if that's super useful for most (80/20) users 🤔 Might be a tad overkill.

Ok, understood.

Edit: One thing that could be helpful on the above, is a swap languages button, that switches the left language to the right and vice versa.

**Comment by azrikahar** [comment]
Mentioning #8203 and #8518 as existing PRs here to prevent potential overlaps 👍 Good news is the former should directly address the 2nd point, whereas the latter is another approach to alleviate some of the other points.

**Reply by bposman** [reply]
> Mentioning #8203 and #8518 as existing PRs here to prevent potential overlaps 👍 Good news is the former should directly address the 2nd point, whereas the latter is another approach to alleviate some of the other points.

Thanks for the heads up on those PRs! 

The first PR is very different from my attempt at this in my poc, so I will review and see if it is a better way of doing it, or if the differences are related to the internal changes of directus since that PR was created.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
